@Test
public void testOnRouteRequestWhenOnNotFoundExceptionInDiagnosticMode() throws Exception {
    FilterChain filterChain = Mockito.mock(FilterChain.class);
    Mockito.when(route.getFilterChain()).thenReturn(filterChain);
    RequestNotFoundException notFoundRequest = new RequestNotFoundException("That's a RequestNotFoundException that should be handled by onNotFound");
    Mockito.when(filterChain.next(contextImpl)).thenThrow(notFoundRequest);
    when(ninjaProperties.isDev()).thenReturn(true);
    when(ninjaProperties.getBooleanWithDefault(NinjaConstant.DIAGNOSTICS_KEY_NAME, true)).thenReturn(true);
    ninjaDefault.onRouteRequest(contextImpl);
    Result localResult = ninjaDefault.getNotFoundResult(contextImpl, notFoundRequest);
    assertThat(localResult.getRenderable(), CoreMatchers.instanceOf(DiagnosticError.class));
}
